packages/node-runtime-worker-thread/src/rpc.ts

```diff
@@ -48,30 +55,27 @@ function send(messageBus: RPCMessageBus, data: any): void {
   }
 }
 
-function serialize(data: unknown): string {
-  return `data:;base64,${v8.serialize(data).toString('base64')}`;
-}
-
-function deserialize<T = unknown>(str: string): T | string {
-  if (/^data:;base64,.+/.test(str)) {
-    return v8.deserialize(
-      Buffer.from(str.replace('data:;base64,', ''), 'base64')
-    );
-  }
-  return str;
-}
-
 function getRPCOptions(messageBus: RPCMessageBus): PostmsgRpcOptions {
   return {
     addListener: messageBus.on.bind(messageBus),
     removeListener: messageBus.off.bind(messageBus),
     postMessage(data) {
       if (isClientMessageData(data) && Array.isArray(data.args)) {
+        // We don't guard against serialization errors on the client, if they
+        // happen, client is responsible for handling them
         data.args = serialize(removeTrailingUndefined(data.args));
       }
 
       if (isServerMessageData(data)) {
-        data.res = serialize(data.res);
+        // If serialization error happened on the server, we use our special
```

Comment:
which server?

Reply:
I think this is clear now, is the part of the RPC protocol that is happening in the child process